Backend/controllers/auth.js

```diff
@@ -35,9 +38,34 @@ export const register = async (req, res, next) => {
  * @throws {Error} - Throws an error if the login is unsuccessful.
  */
 export const login = async (req, res, next) => {
-  // TODO: Implement this function.
   try {
-    return res.send("User logged in");
+    const user = await User.findOne({ email: req.body.email });
+    if (!user) {
+      const error = new Error("Incorrect email or password");
+      error.statusCode = 401;
+      throw error;
+    }
+    const isMatch = await bcrypt.compare(req.body.password, user.password);
+    if (!isMatch) {
+      const error = new Error("Incorrect email or password");
+      error.statusCode = 401;
+      throw error;
+    }
+    const { accessToken, refreshToken } = generateTokens(user._id, user?.roles);
```

Comment:
I used three parameters (email, id and role) to sign the token on /register. you might as well add email to generateTokens()

Reply:
Thank you for this. Let me do the same.